master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go

```diff
@@ -0,0 +1,242 @@
+	k8sV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/determined-ai/determined/master/internal/config"
+	"github.com/determined-ai/determined/master/internal/mocks"
+	"github.com/determined-ai/determined/master/pkg/device"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+const (
+	defaultNamespace    = "default"
+	auxNodeName         = "aux"
+	compNodeName        = "comp"
+	pod1NumSlots        = 4
+	pod2NumSlots        = 8
+	nodeNumSlots        = 8
+	nodeNumSlotsCPU     = 20
+	podServSlotTypeGPU  = "randomDefault"
+	cpuResourceRequests = int64(4000)
+)
+
+// Number of active slots on given nodes.
+var nodeToSlots = map[string]int{
+	auxNodeName:  pod1NumSlots,
+	compNodeName: pod2NumSlots,
+}
+
+func TestGetAgentsAndGetSlots(t *testing.T) {
+	// Create resource list.
+	resourceList := map[k8sV1.ResourceName]resource.Quantity{
+		k8sV1.ResourceName(ResourceTypeNvidia): *resource.NewQuantity(
+			int64(nodeNumSlots),
+			resource.Format("DecimalSI")),
+		k8sV1.ResourceCPU: *resource.NewQuantity(
+			int64(nodeNumSlotsCPU),
+			resource.Format("DecimalSI")),
+	}
+
+	// Create auxiliary agent node and compute node.
+	auxNode := k8sV1.Node{
+		ObjectMeta: metaV1.ObjectMeta{
+			ResourceVersion: "1",
+			Name:            auxNodeName,
+		},
+		Status: k8sV1.NodeStatus{Allocatable: resourceList},
+	}
+
+	compNode := k8sV1.Node{
+		ObjectMeta: metaV1.ObjectMeta{
+			ResourceVersion: "1",
+			Name:            compNodeName,
+		},
+		Status: k8sV1.NodeStatus{Allocatable: resourceList},
+	}
+
+	nodes := map[string]*k8sV1.Node{
+		auxNode.Name:  &auxNode,
+		compNode.Name: &compNode,
+	}
+
+	type AgentsTestCase struct {
+		agentsName   string
+		podsService  *pods
+		wantedAgents map[string]int
+	}
+	tests := []AgentsTestCase{
+		{
+			agentsName:   "GPUPodService",
+			podsService:  createMockPodsService(nodes, podServSlotTypeGPU),
+			wantedAgents: map[string]int{auxNodeName: 0, compNodeName: 0},
+		},
+		{
+			agentsName:   "CUDAPodService",
+			podsService:  createMockPodsService(nodes, device.CUDA),
+			wantedAgents: map[string]int{auxNodeName: 0, compNodeName: 0},
+		},
+		{
+			agentsName:   "CPUPodService",
+			podsService:  createMockPodsService(nodes, device.CPU),
+			wantedAgents: map[string]int{auxNodeName: 0, compNodeName: 0},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.agentsName, func(t *testing.T) {
+			agentsResp := test.podsService.handleGetAgentsRequest()
+			require.Equal(t, len(test.wantedAgents), len(agentsResp.Agents))
+			for _, agent := range agentsResp.Agents {
+				_, ok := test.wantedAgents[agent.Id]
+				require.True(t, ok)
+				agentResp := test.podsService.handleGetAgentRequest(agent.Id)
+				require.Equal(t, agentResp.Agent.Id, agent.Id)
+				slotsResp := test.podsService.handleGetSlotsRequest(agent.Id)
+				require.Equal(t, nodeNumSlots, len(slotsResp.Slots))
+
+				activeSlots := 0
+				cnt := 0
```

Comment:
style: lol can we write this as `count := 0` instead ? I don't think it needs to be abbreviated. Even better, since it doesn't seem like we increment it anywhere, and we only call it twice, I think we can just substitute it in.

Reply:
Ooh good catch! I incremented this initially but somehow removed that from this commit, I'll change the var name to `count` and add back the incrementation.